TrafficCapture/dockerSolution/src/main/docker/migrationConsole/console_api/console_api/apps/orchestrator/views.py

```diff
@@ -37,70 +36,77 @@ def pretty_request(request, data):
-    except InvalidAuthParameters as i:
-        logger.error(f"Error performing osi_create_migration API: {i}")
-        return Response(str(i), status=status.HTTP_400_BAD_REQUEST)
+        osi_action_func(osi_client=determine_osi_client(request_data=request_data, action_name=action_name),
+                        pipeline_name=pipeline_name)
     except Exception as e:
-        logger.error(f"Error performing osi_create_migration API: {e}")
+        logger.error(f'Error performing OSI {action_name} Pipeline API: {e}')
         return Response(str(e), status=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
     response_data = {
-        'Timestamp': datetime.datetime.now(datetime.timezone.utc),
-        'MigrationType': MigrationType.OSI_HISTORICAL_MIGRATION
+        'Timestamp': datetime.datetime.now(datetime.timezone.utc)
     }
     return Response(response_data, status=status.HTTP_200_OK)
 
 
 @api_view(['POST'])
 @parser_classes([JSONParser])
-def osi_start_migration(request):
+def osi_create_migration(request):
     request_data = request.data
     logger.info(pretty_request(request, request_data))
+    action_name = 'Create'
 
-    pipeline_name = request_data.get("PipelineName")
+    osi_serializer = OpenSearchIngestionCreateRequestSerializer(data=request_data)
+    osi_serializer.is_valid(raise_exception=True)
     try:
-        osi_client = boto3.client('osis')
-        start_pipeline(osi_client=osi_client, pipeline_name=pipeline_name)
+        create_pipeline_from_json(osi_client=determine_osi_client(request_data=request_data, action_name=action_name),
+                                  input_json=request_data,
+                                  pipeline_template_path=PIPELINE_TEMPLATE_PATH)
+    except InvalidAuthParameters as i:
+        logger.error(f'Error performing OSI {action_name} Pipeline API: {i}')
+        return Response(str(i), status=status.HTTP_400_BAD_REQUEST)
     except Exception as e:
-        logger.error(f"Error performing osi_start_migration API: {e}")
+        logger.error(f'Error performing OSI {action_name} Pipeline API: {e}')
         return Response(str(e), status=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
     response_data = {
-        'Timestamp': datetime.datetime.now(datetime.timezone.utc),
-        'MigrationType': MigrationType.OSI_HISTORICAL_MIGRATION
+        'Timestamp': datetime.datetime.now(datetime.timezone.utc)
     }
     return Response(response_data, status=status.HTTP_200_OK)
 
 
+@api_view(['POST'])
+@parser_classes([JSONParser])
+def osi_start_migration(request):
+    return osi_update_workflow(request=request, osi_action_func=start_pipeline, action_name='Start')
+
+
 @api_view(['POST'])
 @parser_classes([JSONParser])
 def osi_stop_migration(request):
-    request_data = request.data
-    logger.info(pretty_request(request, request_data))
+    return osi_update_workflow(request=request, osi_action_func=stop_pipeline, action_name='Stop')
```

Comment:
Is there/should there be validation that the pipeline exists for these actions? What happens if it doesn't?

Reply:
I imagine users will get a 500 with the error message, that the pipeline doesn't exist. But agree this is likely a common error case we should test for and be more specific for like a 404 response code. Have created a task around this here: https://opensearch.atlassian.net/browse/MIGRATIONS-1877